3rdparty/tools/rust/BUILD

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+file(
+    name="rustup-init",
+    source=per_platform(
+        linux_arm64=http_source(
+            url="https://static.rust-lang.org/rustup/archive/1.26.0/aarch64-unknown-linux-gnu/rustup-init",
+            len=14131368,
+            sha256="673e336c81c65e6b16dcdede33f4cc9ed0f08bde1dbe7a935f113605292dc800",
+        ),
+        linux_x86_64=http_source(
+            url="https://static.rust-lang.org/rustup/archive/1.26.0/x86_64-unknown-linux-gnu/rustup-init",
+            len=14293176,
+            sha256="0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db",
+        ),
+        macos_arm64=http_source(
+            url="https://static.rust-lang.org/rustup/archive/1.26.0/aarch64-apple-darwin/rustup-init",
+            len=8000713,
+            sha256="ed299a8fe762dc28161a99a03cf62836977524ad557ad70e13882d2f375d3983",
+        ),
+        macos_x86_64=http_source(
+            url="https://static.rust-lang.org/rustup/archive/1.26.0/x86_64-apple-darwin/rustup-init",
+            len=8670640,
+            sha256="f6d1a9fac1a0d0802d87c254f02369a79973bc8c55aa0016d34af4fcdbd67822",
+        ),
+    ),
+)
+
+shell_command(
+    name="cargo-shim",
+    command="""
+        chmod +x rustup-init
+        ./rustup-init --no-modify-path --no-update-default-toolchain --profile minimal --default-toolchain none -y
```

Comment:
FWIW, I could imagine we move towards #17729 to get linting/formatting working, so we might end up switching back to the default profile anyway, and pay that 20%-more-downloads cost.

Reply:
I've had the same imaginative thought. However in those cases, we'd just download the relevant component, no?

Reply:
I'm not sure how independent the components are, eg I _think_ clippy may require rustc, and if we're running them via `cargo ...`, then cargo is needed too... but yeah, we can solve that problem when we get there.

Reply:
Yeah but the local cache is by file, so we wouldn't be wasting anything by splitting them up. But we will be charging extra by not

Reply:
I guess we'd be re-downloading. So network and CPU cost in that regard...

Reply:
buuuut not if we seeded the `_command` for downloading those with this perhaps.

...anywho...